pkgs/by-name/co/container-structure-test/package.nix

```
@@ -1,49 +1,49 @@
 {
   lib,
   stdenv,
-  fetchurl,
+  testers,
+  buildGoModule,
+  fetchFromGitHub,
+  installShellFiles,
+  container-structure-test,
 }:
-let
-  version = "v1.19.1";
-
-  sources = {
-    x86_64-linux = {
-      url = "https://github.com/GoogleContainerTools/container-structure-test/releases/download/${version}/container-structure-test-linux-amd64";
-      hash = "sha256-7KScpej6Km4fuv6XJez4frLwXUm91lII2RLLT71YRrs=";
-    };
-    aarch64-linux = {
-      url = "https://github.com/GoogleContainerTools/container-structure-test/releases/download/${version}/container-structure-test-linux-arm64";
-      hash = "sha256-5MyjOzbDPrV5R3ldJCINipJPOILCzx8+xBVO4bxq9ic=";
-    };
-    x86_64-darwin = {
-      url = "https://github.com/GoogleContainerTools/container-structure-test/releases/download/${version}/container-structure-test-darwin-amd64";
-      hash = "sha256-2QwyKvgzOdGOEuZVwntCpBGBk0JnOLpYOoEYp48qB/I=";
-    };
-    aarch64-darwin = {
-      url = "https://github.com/GoogleContainerTools/container-structure-test/releases/download/${version}/container-structure-test-darwin-arm64";
-      hash = "sha256-x3RmVdDFmHoGOqX49OWeAab/6m1U0jq/g/30rNjj5aI=";
-    };
-  };
-in
-stdenv.mkDerivation {
-  inherit version;
+buildGoModule rec {
+  version = "1.19.3";
   pname = "container-structure-test";
-  src = fetchurl { inherit (sources.${stdenv.system}) url hash; };
+  src = fetchFromGitHub {
+    owner = "GoogleContainerTools";
+    repo = "container-structure-test";
+    rev = "v${version}";
+    sha256 = "sha256-KLLACXUn6dtVCI+gCMHU9hoAJBOAVyhfwxtzsopWS4U=";
+  };
+  vendorHash = "sha256-pBq76HJ+nluOMOs9nqBKp1mr1LuX2NERXo48g8ezE9k=";
 
-  dontUnpack = true;
+  subPackages = [ "cmd/container-structure-test" ];
+  ldflags = [ "-X github.com/${src.owner}/${src.repo}/pkg/version.version=${version}" ];
+  preBuild = ''
+    ldflags+=" -X github.com/${src.owner}/${src.repo}/pkg/version.buildDate=$(date +'%Y-%m-%dT%H:%M:%SZ')"
+  '';
```

```suggestion
  ldflags = [
    "-X github.com/${src.owner}/${src.repo}/pkg/version.version=${version}"
    "-X github.com/${src.owner}/${src.repo}/pkg/version.buildDate=1970-01-01T00:00:00Z"
  ];
```
Otherwise the build is not reproducible.